daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -140,19 +143,36 @@ func New(info logger.Info) (logger.Logger, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	containerStream := &logStream{
 		logStreamName:    logStreamName,
 		logGroupName:     logGroupName,
 		logCreateGroup:   logCreateGroup,
+		logNonBlocking:   logNonBlocking,
 		multilinePattern: multilinePattern,
 		client:           client,
 		messages:         make(chan *logger.Message, 4096),
 	}
-	err = containerStream.create()
-	if err != nil {
-		return nil, err
+
+	creationDone := make(chan bool)
+	if logNonBlocking {
+		go func() {
+			for {
+				err := containerStream.create()
+				if err == nil {
+					break
+				}
```

Comment:
Sleep?

Reply:
I added a 1 second sleep. Do you think this is long enough?

Reply:
We should probably backoff up to some max(30s?)
Also needs to log the errors.

Reply:
Modified it to log the error and to double backoff times until it is > 30s.

Reply:
With doubling each time, 32s would be optimal :)

Reply:
I changed it to 32.